File: LeetCode/Easy/0205-isomorphic-strings/0205-isomorphic-strings.go
func isIsomorphic(s string, t string) bool {
    n := len(s)
    m := make(map[byte]byte)
    x := []byte(t)

    for i := 0; i < n; i++ {
        _, exist := m[s[i]]
        if (!exist) {
            m[s[i]] = t[i]
        }
        x[i] = m[s[i]]
    }
    return t == string(x)
}

File: LeetCode/Easy/0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        Map<Character, Character> trans = new HashMap<>();
        Set<Character> used = new HashSet<>();
        for (int i = 0; i < s.length(); i++) {
            Character sKey = s.charAt(i);
            Character tKey = t.charAt(i);

            if (!trans.containsKey(sKey)) {
                if (used.contains(tKey)) {
                    return false;
                }
                trans.put(sKey, tKey);
                used.add(tKey);
                continue;
            }
            if (trans.get(sKey) != tKey) {
                return false;
            }
        }
        return true;
    }
}